File: src/aws/lambdas/incremental_distribution/src/main/java/uk/nhs/nhsx/core/auth/ResponseSigner.java
package uk.nhs.nhsx.core.auth;

import com.amazonaws.services.lambda.runtime.events.APIGatewayProxyRequestEvent;
import com.amazonaws.services.lambda.runtime.events.APIGatewayProxyResponseEvent;

public interface ResponseSigner {
    void sign(APIGatewayProxyRequestEvent request, APIGatewayProxyResponseEvent response);
}

File: src/aws/lambdas/incremental_distribution/src/main/java/uk/nhs/nhsx/core/auth/ResponseSigner.kt
package uk.nhs.nhsx.core.auth

import com.amazonaws.services.lambda.runtime.events.APIGatewayProxyRequestEvent
import com.amazonaws.services.lambda.runtime.events.APIGatewayProxyResponseEvent

fun interface ResponseSigner {
    fun sign(request: APIGatewayProxyRequestEvent, response: APIGatewayProxyResponseEvent)
}
